### DISCUSSION TITLE
File upload in WebTestCase request results in original file being moved

### DISCUSSION THREAD

**Discussion by samhastings** [symfony/symfony#50652] [Q&A] [answered]
I'm testing an API endpoint that accepts a file upload by creating an instance of `UploadedFile` pointing to an image in my test fixtures directory.
```php
class UploadTest extends WebTestCase
{
    public function testUpload()
    {
        $file = new UploadedFile('/path/to/image.jpg', 'image.jpg');

        static::createClient()->request('POST', '/api/files', files: ['file' => $file]);
    }
}
```

The only problem is the call to `UploadedFile::move()` in my application code is actually moving the original file from my fixtures directory, which currently means I have to re-create the file every time I run the test.

Currently I'm thinking of copying the file to the `sys_get_temp_dir()` directory before uploading, but if there's a better way to avoid the file being moved I'd love to hear it!

Using Symfony 6.2.

**Comment by MatTheCat** [comment, marked as answer]
The file must be moved for the test to work, so the answer is indeed to copy it in a temporary location to be moved from.

See https://symfonycasts.com/screencast/symfony-uploads/fixtures-uploading#copying-the-files-before-moving e.g.

**Comment by samhastings** [comment]
Thanks, assumed this was probably the case but wanted to check just in case there was a better workaround!

**Reply by MatTheCat** [reply]
Yeah I checked before just in case but the screencast was the only reference I found so I guess this is the way to go ¯\\_(ツ)\_/¯
